resources/patches/ruby_27/rbconfig_win.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/rbconfig.rb b/rbconfig.rb
+index de4f173..87459fe 100644
+--- a/rbconfig.rb
++++ b/rbconfig.rb
+@@ -13,7 +13,7 @@ module RbConfig
+     raise "ruby lib version (2.7.8) doesn't match executable version (#{RUBY_VERSION})"
+ 
+   # Ruby installed directory.
+-  TOPDIR = File.dirname(__FILE__).chomp!("/lib/ruby/2.7.0\r/x64-mingw32")
++  TOPDIR = File.dirname(__FILE__).chomp!("/lib/ruby/2.7.0/x64-mingw32")
+   # DESTDIR on make install.
+   DESTDIR = TOPDIR && TOPDIR[/\A[a-z]:/i] || '' unless defined? DESTDIR
+   # The hash configurations stored.
+@@ -54,7 +54,7 @@ module RbConfig
+   CONFIG["sitedir"] = "$(rubylibprefix)/site_ruby"
+   CONFIG["rubyarchdir"] = "$(rubylibdir)/$(arch)"
+   CONFIG["rubylibdir"] = "$(rubylibprefix)/$(ruby_version)"
+-  CONFIG["ruby_version"] = "2.7.0\r"
++  CONFIG["ruby_version"] = "2.7.0"

```

Comment:
I believe `ruby_version` is detected during the ruby build by https://github.com/ruby/ruby/blob/31d0f1a2e7dbfb60731d1f05b868e1d578cda493/configure.ac#L4203 It emits a `conftest.c` program, compiles and runs it, and uses `sed` to extract the version. As mentioned in https://cygwin.com/cygwin-ug-net/using-textbinary.html, cygwin `sed` "Most other programs (such as sed, cmp, tr) use the default mode." The default behavior depends on the filesystem that is mounted. I'm guessing the win2016 cygwin is configured differently than 2012r2.